### PR TITLE
[jsk_perception] apply candidates node supports topic update

### PR DESCRIPTION
**Pull request by knorth55**
## Why?

To change candidates dynamically in multi-object picking challenge https://github.com/start-jsk/jsk_apc/pull/2207.

## Commits

- add `Label` and `LabelArray` msg
- update candidates by topic
  - set param `use_topic` `true` to enable this feature
- update doc

**Updated by @wkentaro**

### CODE REVIEW

File: jsk_perception/node_scripts/apply_context_to_label_probability [hunk 1]
```diff
@@ -49,9 +62,9 @@ class ApplyContextToLabelProbability(ConnectionBasedTransport):
 
         if self.candidates:
             n_labels = proba_img.shape[2]
-            if self.candidates and max(self.candidates) >= n_labels:
```
Comment (wkentaro) [inline jsk_perception/node_scripts/apply_context_to_label_probability]: Why you remove this? `max([])` will raise error.

File: jsk_recognition_msgs/msg/Label.msg [hunk 2]
```diff
@@ -0,0 +1,2 @@
+int32 label
```
Comment (wkentaro) [inline jsk_recognition_msgs/msg/Label.msg]: I thought about for a while last night, and maybe `id` is a better name. What is your idea?

File: jsk_perception/node_scripts/apply_context_to_label_probability (hunk 1 above)
Comment (wkentaro) [inline jsk_perception/node_scripts/apply_context_to_label_probability]: By default, `[]` means all is the candidate.

Comment (knorth55) [inline jsk_perception/node_scripts/apply_context_to_label_probability]: `if self.candidates:` is already checked in upper line.

Comment (wkentaro) [inline jsk_perception/node_scripts/apply_context_to_label_probability]: Oh, I see.

File: jsk_recognition_msgs/msg/Label.msg (hunk 2 above)
Comment (knorth55) [inline jsk_recognition_msgs/msg/Label.msg]: that sounds good to me

**Comment by knorth55** [comment]
@k-okada build fails with this error below.

```

Traceback (most recent call last):
  File "./.travis/travis_jenkins.py", line 367, in <module>
    if j.get_plugin_info('ansicolor'):
  File "/home/travis/.local/lib/python2.7/site-packages/jenkins/__init__.py", line 678, in get_plugin_info
    plugins_info = self.get_plugins(depth)
  File "/home/travis/.local/lib/python2.7/site-packages/jenkins/__init__.py", line 716, in get_plugins
    Request(self._build_url(PLUGIN_INFO, locals()))))
  File "/home/travis/.local/lib/python2.7/site-packages/jenkins/__init__.py", line 430, in jenkins_open
    self.maybe_add_crumb(req)
  File "/home/travis/.local/lib/python2.7/site-packages/jenkins/__init__.py", line 303, in maybe_add_crumb
    self._build_url(CRUMB_URL)), add_crumb=False)
  File "/home/travis/.local/lib/python2.7/site-packages/jenkins/__init__.py", line 448, in jenkins_open
    e.code, e.msg)
jenkins.JenkinsException: Error in request. Possibly authentication failed [401]: Invalid password/token for user: k-okada
+++exit 1
```

**Pull request merged**